plugins/modules/ndo_l3out_node_group_policy.py

```diff
@@ -0,0 +1,316 @@
+short_description: Manage L3Out Node/Interface Group Policy on Cisco Nexus Dashboard Orchestrator (NDO).
+description:
+- Manage L3Out Node/Interface Group Policy on Cisco Nexus Dashboard Orchestrator (NDO).
+- This module is only supported on ND v3.1 (NDO v4.3) and later.
+author:
+- Sabari Jaganathan (@sajagana)
+options:
+  template:
+    description:
+    - The name of the template.
+    - The template must be a L3Out template.
+    type: str
+    required: true
+  l3out:
+    description:
+    - The name of the L3Out.
+    type: str
+    required: true
+  name:
+    description:
+    - The name of the L3Out Node/Interface Group Policy.
+    type: str
+  description:
+    description:
+    - The description of the L3Out Node/Interface Group Policy.
+    type: str
+  node_routing_policy:
+    description:
+    - The name of the L3Out Node Routing Policy.
+    type: str
+  bfd_multi_hop_authentication:
+    description:
+    - The bidirectional forwarding detection (BFD) multi-hop authentication of the L3Out Node/Interface Group Policy.
+    - To enable the O(bfd_multi_hop_authentication) BGP routing protocol must be configured on the L3Out.
+    type: str
+    choices: [ enabled, disabled ]
+  bfd_multi_hop_key_id:
+    description:
+    - The BFD multi-hop key ID of the L3Out Node/Interface Group Policy.
+    type: int
+  bfd_multi_hop_key:
+    description:
+    - The BFD multi-hop key of the L3Out Node/Interface Group Policy.
+    type: str
+  target_dscp:
+    description:
+    - The DSCP Level of the L3Out Node/Interface Group Policy.
+    type: str
+    choices:
+      - af11
+      - af12
+      - af13
+      - af21
+      - af22
+      - af23
+      - af31
+      - af32
+      - af33
+      - af41
+      - af42
+      - af43
+      - cs0
+      - cs1
+      - cs2
+      - cs3
+      - cs4
+      - cs5
+      - cs6
+      - cs7
+      - expedited_forwarding
+      - unspecified
+      - voice_admit
+  state:
+    description:
+    - Use C(absent) for removing.
+    - Use C(query) for listing an object or multiple objects.
+    - Use C(present) for creating or updating.
+    type: str
+    choices: [ absent, query, present ]
+    default: query
+notes:
+- The O(template) must exist before using this module in your playbook.
+  Use M(cisco.mso.ndo_template) to create the L3Out template.
+- The O(l3out) must exist before using this module in your playbook.
+  Use M(cisco.mso.ndo_l3out_template) to create the L3Out object under the L3Out template.
+- The O(node_routing_policy) must exist before using this module in your playbook.
+  Use M(cisco.mso.ndo_l3out_node_routing_policy) to create the L3Out Node Routing Policy.
+extends_documentation_fragment: cisco.mso.modules
+"""
+
+EXAMPLES = r"""
+- name: Create a new L3Out node group policy
+  cisco.mso.ndo_l3out_node_group_policy:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: l3out_template
+    l3out: l3out
+    name: "node_group_policy_1"
```

Comment:
why use quotes for some inputs? think we can remove in examples
```suggestion
    name: node_group_policy_1
```